### PR TITLE
Rewrite the Weather to an Edge App

### PR DESCRIPTION
**Pull request by nicomiguelino**
#### General

- Resolves Screenly/Playground#99

#### Checklist

- [x] Move contents of `edge-apps/weather/src` into `edge-apps/weather` (the root directory) instead.
- [x] Remove unnecessary files &mdash; `gulpfile.js`, `.babelrc`, `load-metadata.js`, `load-screenly.js`.
- [x] Modify `index.html`.
- [x] Make use of client-side storage for caching the API key and weather data instead.
- [x] Have `main.js` get the OpenWeatherMap API key from the settings instead.

#### Screenshots

In order to test the Weather app (rewritten as an Edge App), started a simple HTTP server by running `python3 -m http.server`, which hosts the app at `0.0.0.0:8000` by default. I've also used a mocked `screenly.js`, which I didn't commit in this PR.

image

### CODE REVIEW

File: edge-apps/weather/index.html [hunk 1]
```diff
@@ -48,24 +48,40 @@
...
+            integrity="sha512-42PE0rd+wZ2hNXftlM78BSehIGzezNeQuzihiBCvUEB3CVxHvsShF86wBWwQORNxNINlBPuq7rG4WWhNiTVHFg=="
+            crossorigin="anonymous"
+            referrerpolicy="no-referrer"
+        ></script>
         <script src="/static/js/icons.js"></script>
-        <script src="/static/js/main.js"></script>
-        <script src="/static/js/load-metadata.js"></script>
-        <script src="/static/js/load-screenly.js"></script>
+        <script src="/static/js/main.js">
+            console.log('Screenly OSE Weather App')
```
Comment (iamplugged) [inline edge-apps/weather/index.html:83]: Does this work? There's an src specified along with inline script within the script tags.

Comment (nicomiguelino) [reply edge-apps/weather/index.html:83]: Thanks for pointing this one out. I tried it again, and yes, it doesn't work. I'll go remove it in the succeeding commits.

Comment (nicomiguelino) [reply edge-apps/weather/index.html:83]: Just got back here after addressing code review comments for the Clock migration. My bad, that's just a temporary change.

**Comment by vpetersson** [comment]
@nicomiguelino what are these things? Is this an encoding issue?
Screenshot 2023-08-14 at 10 55 56 AM

**Review by vpetersson — APPROVED** [review]
Approved pending question about encoding issue.

File: edge-apps/weather/index.html [hunk 2]
```diff
@@ -41,31 +41,48 @@
...
+
+                gtag('js', new Date());
+                gtag('config', gaApiKey);
+            }
         </script>
-        <!--endRemoveIf(googleAnalytics)-->
-        <script src="https://cdnjs.cloudflare.com/ajax/libs/moment.js/2.29.4/moment-with-locales.min.js" integrity="sha512-42PE0rd+wZ2hNXftlM78BSehIGzezNeQuzihiBCvUEB3CVxHvsShF86wBWwQORNxNINlBPuq7rG4WWhNiTVHFg==" crossorigin="anonymous" referrerpolicy="no-referrer"></script>
+
+        <script
+            src="https://cdnjs.cloudflare.com/ajax/libs/moment.js/2.29.4/moment-with-locales.min.js"
```
Comment (vpetersson) [inline edge-apps/weather/index.html:80]: Let's ship moment as part of the edge app.

File: edge-apps/weather/static/js/main.js [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+    transaction.addEventListener('error', () => {
+      console.error('Transaction not opened due to error')
+    })
+  }
+
+  const fetchWeather = async () => {
+    try {
+      const endpointUrl = `https://api.openweathermap.org/data/2.5/forecast`
+      const apiKey = screenly.settings.openweathermap_api_key
+      const queryParams = `lat=${lat}&lon=${lng}&units=metric&cnt=10&appid=${apiKey}`
```
Comment (vpetersson) [inline edge-apps/weather/static/js/main.js:289]: Maybe it's easier to just pass the right unit here (`matric` is hard coded) than to do Fahrenheit conversion.

**Comment by nicomiguelino** [comment]
> @nicomiguelino what are these things? Is this an encoding issue? Screenshot 2023-08-14 at 10 55 56 AM

My bad, I haven't updated the screenshots. It's all good now.

File: edge-apps/weather/index.html (hunk 2 above)
Comment (nicomiguelino) [reply edge-apps/weather/index.html:80]: All done (locally). Expect changes on succeeding commit/s.

File: edge-apps/weather/static/js/main.js (hunk 3 above)
Comment (nicomiguelino) [reply edge-apps/weather/static/js/main.js]: I'll address this comment on another issue or pull request. I'll keep you posted.

**Comment by nicomiguelino** [comment]
@vpetersson, I'd like to request a code review. Thanks.

**Review by vpetersson — APPROVED** [review]

**Pull request merged**